### DISCUSSION TITLE
H3 geohashes and Uxarray ... is there a match in heaven?

### DISCUSSION THREAD

**Discussion by dev360** [UXARRAY/uxarray#218] [Q&A] [answered]
Hello,

First off, let me preface my question by saying I have working knowledge of xarray but data science and GIS is not my specialty.

We are prototyping different ways of solving a geographical intersection problem, and our team has been looking at a few different solutions. Its essentially a lat/long grid space, so we looked at geohashing, and H3 in particular appeals to us.

So we essentially have potentially thousands of variables across a x/y grid space, and each variable is a sparse matrix (filled 2-3%, e.g. country sized at most). When we walk down this path, the obvious thing you realize with H3 is that a state like Texas would be like 300k cells at resolution 10 (5km), but would only be 5300 something cells when you collapse it and express it as a combination of resulting resolution 1 through 10 cells.

My question is this: would uxarray be the right tool for the job for this? We want to essentially be able to query for the intersection (values of all matrices) at a given x/y level, expressed at the highest resolution (10 in example above).

Here is a drawing of it for the visually inclined:

<img width="1418" alt="Screenshot 2023-02-02 at 7 16 59 PM" src="https://user-images.githubusercontent.com/291749/216480227-555cb30d-418b-444f-812b-4992931330c0.png">

**Comment by philipc2** [comment]
Hi @dev360 

This is some very interesting work! I don't believe that `uxarray` would be the best fit for this task since we are focused around extending `xarray` to support unstructured grids and this problem appears to be dependent on high-performance GIS algorithms and data structures.

Since you are already familiar with `xarray`, I would take a look at [`xarray-spatial`](https://xarray-spatial.org/). Another package worth looking into (especially if this problem can be GPU accelerated) would be [`cuSpatial`](https://docs.rapids.ai/api/cuspatial/stable/). 

Hope this helps!

**Reply by dev360** [reply]
Interesting. Yeah the first thing to came to my mind was actually GPU/Cuda, but there is no appetite for very low-level solution (Rust). The tooling is painful on Mac too since AWS only supports Nvidia. I'll def check out the libraries you suggested! Thanks

**Comment by dcherian** [comment, marked as answer]
I would ask over at https://github.com/martinfleis/xvec
